cloudinit/config/schema.py

```diff
@@ -1002,7 +1098,14 @@ def validate_cloudconfig_file(
         if not netcfg:
             print("Skipping network-config schema validation on empty config.")
             return False
-        elif netcfg.get("version") != 1:
+        elif netcfg.get("version") == 2:
```

Comment:
Good point I refactored netplan_validate_network_config to accept just the network_config dict instead of dealing with files. This simplified callsites sigificantly and allowed dealing with version 2 vs version 1 vs netplan API availability inside `netplan_validate_network_schema` function. If unable to perform netplan_validate.... we fallthrough and use cloud-init's schema for validation if present. Since we don't have JSON schema version 2 yet, we will still emit Skipping schema validation messages when not netplan and not network schema `version: 1`.

Reply:
+1 I was avoiding the refactor to validate_cloudconfig_schema that would require adding a config path. But, I've since refactored this code to simplify call-sites and like cloudinit.stages